tensorflow_model_optimization/g3doc/guide/clustering/index.md

```diff
@@ -0,0 +1,99 @@
+
+Clustering works well with TFLiteConverter, providing an easy path to produce deployment-ready models that can be easily compressed using either an off-the-shelf compression algorithm, similar to the ZIP Deflate we use for demonstration in this document, or a custom method optimized for a special target hardware. When converting the clustered model with TFLiteConverter, the actual number of unique weight values per tensor may increase. This happens for the models with batch normalization layers that are folded into the preceding convolutional layers during the conversion, and also due to different scale factors in the per-channel weight quantization scheme. Both techniques may alter the same weight value differently, depending on the channel it appears in and the associated batch-normalization and quantization parameters. While this side effect may result in a slightly lower compression ratio, the overall benefits of using clustering and post-training conversion and quantization are still tangible, as demonstrated by the examples in this document.
+
+### API compatibility matrix
+
+Users can apply clustering with the following APIs:
+
+*   Model building: `tf.keras` with only Sequential and Functional models
+*   TensorFlow versions: TF 1.x for versions 1.14+ and 2.x.
+    *   `tf.compat.v1` with a TF 2.X package and `tf.compat.v2` with a TF 1.X
+        package are not supported.
+*   TensorFlow execution mode: both graph and eager
+
+## Results
+
+### Image classification
+
+<table>
+  <tr>
+    <th rowspan=2>Model</th>
+    <th colspan=2>Original</th>
+    <th colspan=3>Clustered</th>
+  </tr>
+  <tr>
+    <th>Top-1 accuracy</th>
+    <th>Size of compressed .tflite</th>
+    <th># of clusters</th>
+    <th>Top-1 accuracy</th>
+    <th>Size of compressed .tflite</th>
+  </tr>
+  <tr>
+    <td>MobileNetV2</td>
+    <td>72.29%</td>
+    <td>13.0 MB</td>
+    <td>32</td>
+    <td>69.33%</td>
+    <td>2.6 MB</td>
+  </tr>
+</table>
+
+The models were trained and tested on ImageNet.
+
+### Keyword spotting
+
+<table>
+  <tr>
+    <th rowspan=2>Model</th>
+    <th colspan=2>Original</th>
+    <th colspan=3>Clustered</th>
+  </tr>
+  <tr>
+    <th>Top-1 accuracy</th>
+    <th>Size of compressed .tflite</th>
+    <th># of clusters</th>
+    <th>Top-1 accuracy</th>
+    <th>Size of compressed .tflite</th>
+  </tr>
+  <tr>
+    <td>DS-CNN-L</td>
+    <td>95.03%</td>
+    <td>1.5 MB</td>
+    <td>32</td>
+    <td>94.71%</td>
+    <td>0.3 MB</td>
+  </tr>
+</table>
+
+The models were trained and tested on SpeechCommands v0.02.
+
+NOTE: *Size of compressed .tflite* refers to the size of the zipped .tflite file obtained from the model through the following process:
```

Comment:
Edit for conciseness: "file obtained from the model through the following process" -> "file obtained from the following process"